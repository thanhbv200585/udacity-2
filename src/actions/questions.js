import { saveQuestionAnswer, saveQuestion } from "../utils/api";
import {
  addQuestionToAuthor,
  addAnsweredQuestionToUser,
} from "../actions/users";
import { showLoading, hideLoading } from "react-redux-loading-bar";

export const RECEIVE_QUESTIONS = "RECEIVE_QUESTIONS";
export const ANSWER_QUESTION = "ANSWER_QUESTION";
export const ADD_QUESTION = "ADD_QUESTION";

export function receiveQuestions(questions) {
  return {
    type: RECEIVE_QUESTIONS,
    questions,
  };
}

function answerQuestion({ authedUser, qid, answer }) {
  return {
    type: ANSWER_QUESTION,
    authedUser,
    qid,
    answer,
  };
}

export function handleAnswerQuestion(info) {
  return (dispatch) => {
    dispatch(showLoading());
    dispatch(answerQuestion(info));
    dispatch(addAnsweredQuestionToUser(info));
    dispatch(hideLoading());

    return saveQuestionAnswer(info).catch((e) => {
      console.warn("Error in saveQuestionAnswer: ", e);
      alert("There was an error. Try again");
    });
  };
}

function addQuestion(question) {
  return {
    type: ADD_QUESTION,
    question,
  };
}

export function handleAddQuestion(optionOne, optionTwo) {
  return (dispatch, getState) => {
    const { authedUser } = getState();
    return saveQuestion({
      optionOneText: optionOne,
      optionTwoText: optionTwo,
      author: authedUser,
    }).then((question) => {
      dispatch(showLoading());
      dispatch(addQuestion(question));
      dispatch(addQuestionToAuthor(question));
      dispatch(hideLoading());
    })
    .catch((e) => {
      console.warn("Error in saveQuestion: ", e);
      alert("There was an error. Try again");
    });
  };
}
