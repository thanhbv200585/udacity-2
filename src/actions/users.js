export const RECEIVE_USERS = "RECEIVE_USERS";
export const ADD_ANSWERED_QUESTION = "ADD_ANSWERED_QUESTION";
export const ADD_QUESTION_TO_AUTHOR = "ADD_QUESTION_TO_AUTHOR";

export function receiveUsers(users) {
  return {
    type: RECEIVE_USERS,
    users,
  };
}

export function addAnsweredQuestionToUser({ qid, authedUser, answer }) {
  return {
    type: ADD_ANSWERED_QUESTION,
    qid,
    authedUser,
    answer,
  };
}

export function addQuestionToAuthor(question) {
  return {
    type: ADD_QUESTION_TO_AUTHOR,
    question,
  };
}
