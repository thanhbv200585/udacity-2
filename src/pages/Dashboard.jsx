import NewQuestions from "../components/NewQuestions";
import AnsweredQuestions from "../components/AnsweredQuestions";
import { useState } from "react";
import clsx from "clsx";

const TABS = ["New Questions", "Answered Questions"];

const Dashboard = () => {
  const [showing, setShowing] = useState("New Questions");
  const setTab = (tab) => {
    setShowing(tab);
  };

  return (
    <div className="max-w-3xl w-full mx-auto mt-6">
      <div className="flex items-stretch overflow-hidden">
        {TABS.map((tab) => (
          <button
            key={tab}
            className={clsx(
              "flex-1 p-4 transition-all border border-solid border-[--divider] rounded-tl-md rounded-tr-md ",
              tab === showing ? "border-b-0 bg-white" : 'bg-[--gray-40]'
            )}
            onClick={() => setTab(tab)}
          >
            {tab}
          </button>
        ))}
      </div>
      <div className="p-4 border border-solid border-[--divider] border-t-0 rounded-bl-md rounded-br-md">
        {showing === "New Questions" && <NewQuestions />}
        {showing === "Answered Questions" && <AnsweredQuestions />}
      </div>
    </div>
  );
};

export default Dashboard;
