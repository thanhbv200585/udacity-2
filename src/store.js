import reducer from "./reducers";
import { createStore } from "redux";
import middleware from "./middleware";

const store = createStore(reducer, middleware);

export { store };
