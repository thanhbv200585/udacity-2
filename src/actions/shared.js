import { getInitialData } from "../utils/api";
import { receiveQuestions } from "./questions";
import { receiveUsers } from "./users";
import { setAuthedUser } from "./authedUser";
import { showLoading, hideLoading } from "react-redux-loading-bar";

const AUTHED_ID = null

export function handleInitialData() {
  return (dispatch) => {
    dispatch(showLoading());
    return getInitialData().then(({ users, questions }) => {
      dispatch(receiveQuestions(questions));
      dispatch(receiveUsers(users));
      dispatch(setAuthedUser(AUTHED_ID));
      dispatch(hideLoading());
    });
  };
}
