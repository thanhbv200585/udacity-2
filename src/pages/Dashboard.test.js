import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Dashboard from "./Dashboard";
import { store } from "../store";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { setAuthedUser } from "../actions/authedUser";
import { handleInitialData } from "../actions/shared";

describe("Dashboard", () => {
  it("Is expected to load the new and answered polls.", async () => {
    await store.dispatch(setAuthedUser("mtsamis"));
    await store.dispatch(handleInitialData());
    const view = render(
      <Provider store={store}>
        <Router>
          <Dashboard />
        </Router>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});
