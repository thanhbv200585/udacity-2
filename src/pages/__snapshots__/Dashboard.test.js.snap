// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Is expected to load the new and answered polls. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-3xl w-full mx-auto mt-6"
      >
        <div
          class="flex items-stretch overflow-hidden"
        >
          <button
            class="flex-1 p-4 transition-all border border-solid border-[--divider] rounded-tl-md rounded-tr-md  border-b-0 bg-white"
          >
            New Questions
          </button>
          <button
            class="flex-1 p-4 transition-all border border-solid border-[--divider] rounded-tl-md rounded-tr-md  bg-[--gray-40]"
          >
            Answered Questions
          </button>
        </div>
        <div
          class="p-4 border border-solid border-[--divider] border-t-0 rounded-bl-md rounded-br-md"
        >
          <div>
            <ul
              class="flex flex-col gap-4"
            >
              <li>
                <div
                  class="shadow-md"
                >
                  <p
                    class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                  >
                    Mike Tsamis
                  </p>
                  <div
                    class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                  >
                    <img
                      alt="Avatar of Mike Tsamis"
                      class="w-32 h-32"
                      src="https://avatar.iran.liara.run/username?username=Mike+Tsamis"
                    />
                    <span
                      class="w-px bg-[--divider]"
                    />
                    <div
                      class="flex-1 flex flex-col"
                    >
                      <h2
                        class="font-semibold"
                      >
                        Would you rather
                      </h2>
                      <div
                        class="flex-1 flex flex-col justify-center items-center"
                      >
                        <p>
                          deploy to production once every two weeks
                        </p>
                        <span>
                          or...
                        </span>
                      </div>
                      <a
                        href="/questions/xj352vofupe1dqz9emx13r"
                      >
                        <button
                          class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                          type="button"
                        >
                          Answer Poll
                        </button>
                      </a>
                    </div>
                  </div>
                </div>
              </li>
              <li>
                <div
                  class="shadow-md"
                >
                  <p
                    class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                  >
                    Tyler McGinnis
                  </p>
                  <div
                    class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                  >
                    <img
                      alt="Avatar of Tyler McGinnis"
                      class="w-32 h-32"
                      src="https://avatar.iran.liara.run/username?username=Tyler+McGinnis"
                    />
                    <span
                      class="w-px bg-[--divider]"
                    />
                    <div
                      class="flex-1 flex flex-col"
                    >
                      <h2
                        class="font-semibold"
                      >
                        Would you rather
                      </h2>
                      <div
                        class="flex-1 flex flex-col justify-center items-center"
                      >
                        <p>
                          take a course on ReactJS
                        </p>
                        <span>
                          or...
                        </span>
                      </div>
                      <a
                        href="/questions/vthrdm985a262al8qx3do"
                      >
                        <button
                          class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                          type="button"
                        >
                          Answer Poll
                        </button>
                      </a>
                    </div>
                  </div>
                </div>
              </li>
              <li>
                <div
                  class="shadow-md"
                >
                  <p
                    class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                  >
                    Sarah Edo
                  </p>
                  <div
                    class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                  >
                    <img
                      alt="Avatar of Sarah Edo"
                      class="w-32 h-32"
                      src="https://avatar.iran.liara.run/username?username=Sarah+Edo"
                    />
                    <span
                      class="w-px bg-[--divider]"
                    />
                    <div
                      class="flex-1 flex flex-col"
                    >
                      <h2
                        class="font-semibold"
                      >
                        Would you rather
                      </h2>
                      <div
                        class="flex-1 flex flex-col justify-center items-center"
                      >
                        <p>
                          conduct a release retrospective 1 week after a release
                        </p>
                        <span>
                          or...
                        </span>
                      </div>
                      <a
                        href="/questions/am8ehyc8byjqgar0jgpub9"
                      >
                        <button
                          class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                          type="button"
                        >
                          Answer Poll
                        </button>
                      </a>
                    </div>
                  </div>
                </div>
              </li>
              <li>
                <div
                  class="shadow-md"
                >
                  <p
                    class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                  >
                    Tyler McGinnis
                  </p>
                  <div
                    class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                  >
                    <img
                      alt="Avatar of Tyler McGinnis"
                      class="w-32 h-32"
                      src="https://avatar.iran.liara.run/username?username=Tyler+McGinnis"
                    />
                    <span
                      class="w-px bg-[--divider]"
                    />
                    <div
                      class="flex-1 flex flex-col"
                    >
                      <h2
                        class="font-semibold"
                      >
                        Would you rather
                      </h2>
                      <div
                        class="flex-1 flex flex-col justify-center items-center"
                      >
                        <p>
                          have code reviews conducted by peers
                        </p>
                        <span>
                          or...
                        </span>
                      </div>
                      <a
                        href="/questions/loxhs1bqm25b708cmbf3g"
                      >
                        <button
                          class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                          type="button"
                        >
                          Answer Poll
                        </button>
                      </a>
                    </div>
                  </div>
                </div>
              </li>
              <li>
                <div
                  class="shadow-md"
                >
                  <p
                    class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                  >
                    Mike Tsamis
                  </p>
                  <div
                    class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                  >
                    <img
                      alt="Avatar of Mike Tsamis"
                      class="w-32 h-32"
                      src="https://avatar.iran.liara.run/username?username=Mike+Tsamis"
                    />
                    <span
                      class="w-px bg-[--divider]"
                    />
                    <div
                      class="flex-1 flex flex-col"
                    >
                      <h2
                        class="font-semibold"
                      >
                        Would you rather
                      </h2>
                      <div
                        class="flex-1 flex flex-col justify-center items-center"
                      >
                        <p>
                          hire more frontend developers
                        </p>
                        <span>
                          or...
                        </span>
                      </div>
                      <a
                        href="/questions/6ni6ok3ym7mf1p33lnez"
                      >
                        <button
                          class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                          type="button"
                        >
                          Answer Poll
                        </button>
                      </a>
                    </div>
                  </div>
                </div>
              </li>
              <li>
                <div
                  class="shadow-md"
                >
                  <p
                    class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                  >
                    Sarah Edo
                  </p>
                  <div
                    class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                  >
                    <img
                      alt="Avatar of Sarah Edo"
                      class="w-32 h-32"
                      src="https://avatar.iran.liara.run/username?username=Sarah+Edo"
                    />
                    <span
                      class="w-px bg-[--divider]"
                    />
                    <div
                      class="flex-1 flex flex-col"
                    >
                      <h2
                        class="font-semibold"
                      >
                        Would you rather
                      </h2>
                      <div
                        class="flex-1 flex flex-col justify-center items-center"
                      >
                        <p>
                          Build our new application with Javascript
                        </p>
                        <span>
                          or...
                        </span>
                      </div>
                      <a
                        href="/questions/8xf0y6ziyjabvozdd253nd"
                      >
                        <button
                          class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                          type="button"
                        >
                          Answer Poll
                        </button>
                      </a>
                    </div>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-3xl w-full mx-auto mt-6"
    >
      <div
        class="flex items-stretch overflow-hidden"
      >
        <button
          class="flex-1 p-4 transition-all border border-solid border-[--divider] rounded-tl-md rounded-tr-md  border-b-0 bg-white"
        >
          New Questions
        </button>
        <button
          class="flex-1 p-4 transition-all border border-solid border-[--divider] rounded-tl-md rounded-tr-md  bg-[--gray-40]"
        >
          Answered Questions
        </button>
      </div>
      <div
        class="p-4 border border-solid border-[--divider] border-t-0 rounded-bl-md rounded-br-md"
      >
        <div>
          <ul
            class="flex flex-col gap-4"
          >
            <li>
              <div
                class="shadow-md"
              >
                <p
                  class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                >
                  Mike Tsamis
                </p>
                <div
                  class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                >
                  <img
                    alt="Avatar of Mike Tsamis"
                    class="w-32 h-32"
                    src="https://avatar.iran.liara.run/username?username=Mike+Tsamis"
                  />
                  <span
                    class="w-px bg-[--divider]"
                  />
                  <div
                    class="flex-1 flex flex-col"
                  >
                    <h2
                      class="font-semibold"
                    >
                      Would you rather
                    </h2>
                    <div
                      class="flex-1 flex flex-col justify-center items-center"
                    >
                      <p>
                        deploy to production once every two weeks
                      </p>
                      <span>
                        or...
                      </span>
                    </div>
                    <a
                      href="/questions/xj352vofupe1dqz9emx13r"
                    >
                      <button
                        class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                        type="button"
                      >
                        Answer Poll
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </li>
            <li>
              <div
                class="shadow-md"
              >
                <p
                  class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                >
                  Tyler McGinnis
                </p>
                <div
                  class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                >
                  <img
                    alt="Avatar of Tyler McGinnis"
                    class="w-32 h-32"
                    src="https://avatar.iran.liara.run/username?username=Tyler+McGinnis"
                  />
                  <span
                    class="w-px bg-[--divider]"
                  />
                  <div
                    class="flex-1 flex flex-col"
                  >
                    <h2
                      class="font-semibold"
                    >
                      Would you rather
                    </h2>
                    <div
                      class="flex-1 flex flex-col justify-center items-center"
                    >
                      <p>
                        take a course on ReactJS
                      </p>
                      <span>
                        or...
                      </span>
                    </div>
                    <a
                      href="/questions/vthrdm985a262al8qx3do"
                    >
                      <button
                        class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                        type="button"
                      >
                        Answer Poll
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </li>
            <li>
              <div
                class="shadow-md"
              >
                <p
                  class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                >
                  Sarah Edo
                </p>
                <div
                  class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                >
                  <img
                    alt="Avatar of Sarah Edo"
                    class="w-32 h-32"
                    src="https://avatar.iran.liara.run/username?username=Sarah+Edo"
                  />
                  <span
                    class="w-px bg-[--divider]"
                  />
                  <div
                    class="flex-1 flex flex-col"
                  >
                    <h2
                      class="font-semibold"
                    >
                      Would you rather
                    </h2>
                    <div
                      class="flex-1 flex flex-col justify-center items-center"
                    >
                      <p>
                        conduct a release retrospective 1 week after a release
                      </p>
                      <span>
                        or...
                      </span>
                    </div>
                    <a
                      href="/questions/am8ehyc8byjqgar0jgpub9"
                    >
                      <button
                        class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                        type="button"
                      >
                        Answer Poll
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </li>
            <li>
              <div
                class="shadow-md"
              >
                <p
                  class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                >
                  Tyler McGinnis
                </p>
                <div
                  class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                >
                  <img
                    alt="Avatar of Tyler McGinnis"
                    class="w-32 h-32"
                    src="https://avatar.iran.liara.run/username?username=Tyler+McGinnis"
                  />
                  <span
                    class="w-px bg-[--divider]"
                  />
                  <div
                    class="flex-1 flex flex-col"
                  >
                    <h2
                      class="font-semibold"
                    >
                      Would you rather
                    </h2>
                    <div
                      class="flex-1 flex flex-col justify-center items-center"
                    >
                      <p>
                        have code reviews conducted by peers
                      </p>
                      <span>
                        or...
                      </span>
                    </div>
                    <a
                      href="/questions/loxhs1bqm25b708cmbf3g"
                    >
                      <button
                        class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                        type="button"
                      >
                        Answer Poll
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </li>
            <li>
              <div
                class="shadow-md"
              >
                <p
                  class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                >
                  Mike Tsamis
                </p>
                <div
                  class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                >
                  <img
                    alt="Avatar of Mike Tsamis"
                    class="w-32 h-32"
                    src="https://avatar.iran.liara.run/username?username=Mike+Tsamis"
                  />
                  <span
                    class="w-px bg-[--divider]"
                  />
                  <div
                    class="flex-1 flex flex-col"
                  >
                    <h2
                      class="font-semibold"
                    >
                      Would you rather
                    </h2>
                    <div
                      class="flex-1 flex flex-col justify-center items-center"
                    >
                      <p>
                        hire more frontend developers
                      </p>
                      <span>
                        or...
                      </span>
                    </div>
                    <a
                      href="/questions/6ni6ok3ym7mf1p33lnez"
                    >
                      <button
                        class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                        type="button"
                      >
                        Answer Poll
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </li>
            <li>
              <div
                class="shadow-md"
              >
                <p
                  class="border border-solid border-[--divider] rounded-tl-md rounded-tr-md p-3 bg-[--gray-40] border-t-2 border-t-yellow-700"
                >
                  Sarah Edo
                </p>
                <div
                  class="border border-solid border-[--divider] border-t-0 p-3 rounded-bl-md rounded-br-md flex flex items-stretch gap-4"
                >
                  <img
                    alt="Avatar of Sarah Edo"
                    class="w-32 h-32"
                    src="https://avatar.iran.liara.run/username?username=Sarah+Edo"
                  />
                  <span
                    class="w-px bg-[--divider]"
                  />
                  <div
                    class="flex-1 flex flex-col"
                  >
                    <h2
                      class="font-semibold"
                    >
                      Would you rather
                    </h2>
                    <div
                      class="flex-1 flex flex-col justify-center items-center"
                    >
                      <p>
                        Build our new application with Javascript
                      </p>
                      <span>
                        or...
                      </span>
                    </div>
                    <a
                      href="/questions/8xf0y6ziyjabvozdd253nd"
                    >
                      <button
                        class="text-white focus:outline-none w-full focus:ring-4 font-medium rounded-lg text-sm px-5 py-2.5 disabled:bg-gray-500 bg-yellow-700 hover:bg-yellow-800 focus:ring-yellow-400"
                        type="button"
                      >
                        Answer Poll
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
